### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by whitedevil-source**

### CODE REVIEW

File: MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  }
+  getShow(): Visibility{
+    return this.show
+  }
+  getImage():Resource{
+    return this.image
+  }
+
+  setButtonColor(value:string|ResourceColor):BannerModel{
+    if(value==null || value==undefined){
```
Comment (HMOS-Developer) [inline MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets]: null validations are not properly handled

File: MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      console.log("Button color not set. Default value used")
+    }
+    this.buttonColor=value
+    return this
+  }
+  setTextSize(tc:string):BannerModel{
+    if(tc==null || tc==undefined){
```
Comment (HMOS-Developer) [inline MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets]: null validations are not properly handled

File: MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      console.log("Text size not set. Default value used")
+    }
+    this.textSize=tc
+    return this
+  }
+  setTextStyle(ts:string):BannerModel{
+    if(ts==null || ts==undefined){
```
Comment (HMOS-Developer) [inline MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets]: null validations are not properly handled

File: MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      console.log("Text style not set. Default value used")
+    }
+    this.textStyle=ts
+    return this
+  }
+  setBannerMessage(s:string):BannerModel{
+    if(s==null || s==undefined){
```
Comment (HMOS-Developer) [inline MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets]: null validations are not properly handled

File: MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      console.log("Banner Message not set")
+    }
+    this.bannerMessage=s
+    return this
+  }
+  setButtonText1(a1:string):BannerModel{
+    if(a1==null || a1==undefined){
```
Comment (HMOS-Developer) [inline MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets]: null validations are not properly handled

File: MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+      console.log("Button Text 1 not set")
+    }
+    this.buttonText1=a1
+    return this
+  }
+  setButtonText2(a2:string):BannerModel{
+    if(this.bannerType==BannerType.TwoButton && (a2==null || a2==undefined)){
```
Comment (HMOS-Developer) [inline MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets]: null validations are not properly handled

File: MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+      console.log("Button Text 2 not set")
+    }
+    if(this.bannerType==BannerType.ImageButton && (a2==null || a2==undefined)){
+      console.log("Button Text 2 not set")
+    }
+    this.buttonText2=a2
+    return this
+  }
+  setImage(img:Resource):BannerModel{
+    if(this.bannerType==BannerType.ImageButton && (img==null || img==undefined)){
```
Comment (HMOS-Developer) [inline MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets]: null validations are not properly handled

File: MaterialBanner/src/main/ets/components/MainPage/MainPage.ets [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {BannerModel,BannerType} from './BannerModel'
+
+@Component
+export struct MaterialBanner{
+  @State bModel:BannerModel=new BannerModel(BannerType.OneButton);
+  buttonAction1:()=>void;
```
Comment (HMOS-Developer) [inline MaterialBanner/src/main/ets/components/MainPage/MainPage.ets]: rename buttonAction1 variable to something more user understandable

File: MaterialBanner/src/main/ets/components/MainPage/MainPage.ets [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    Column(){
+      if(this.bModel.getBannerType()==BannerType.OneButton){
+        Column() {
+          Flex({ justifyContent: FlexAlign.SpaceBetween }) {
+            Text(this.bModel.getBannerMessage()).padding({ top: '10vp', bottom: '8vp', left: '20vp' })
+              .fontSize(this.bModel.getTextSize()).fontFamily(this.bModel.getTextStyle())
+            Button(this.bModel.getButtonText1(), { type: ButtonType.Normal, stateEffect: true }).borderRadius('8vp')
+              .backgroundColor(this.bModel.getButtonColor())
+              .onClick(()=>{
+                this.buttonAction1()
```
Comment (HMOS-Developer) [inline MaterialBanner/src/main/ets/components/MainPage/MainPage.ets]: handle null & undefined validation for buttonAction1 callback function

File: MaterialBanner/src/main/ets/components/MainPage/MainPage.ets [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+      else if(this.bModel.getBannerType()==BannerType.TwoButton){
+        Column() {
+          Text(this.bModel.getBannerMessage())
+            .fontSize(this.bModel.getTextSize()).fontFamily(this.bModel.getTextStyle())
+            .padding({ top: '36vp', left: '16vp' }).width('100%')
+          Flex({ direction: FlexDirection.RowReverse, alignItems: ItemAlign.End, justifyContent: FlexAlign.SpaceBetween }) {
+            Button(this.bModel.getButtonText2(), { type: ButtonType.Normal, stateEffect: true }).borderRadius('5vp')
+              .backgroundColor(this.bModel.getButtonColor())
+              .onClick((event: ClickEvent) => {
+                this.buttonAction2()
```
Comment (HMOS-Developer) [inline MaterialBanner/src/main/ets/components/MainPage/MainPage.ets]: handle null & undefined validation for buttonAction2 callback function

File: MaterialBanner/src/main/ets/components/MainPage/MainPage.ets [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+                this.bModel.setShow(Visibility.None)
+              })
+            Button(this.bModel.getButtonText1(), { type: ButtonType.Normal, stateEffect: true }).borderRadius('5vp')
+              .backgroundColor(this.bModel.getButtonColor())
+              .onClick((event: ClickEvent) => {
+                this.buttonAction1()
```
Comment (HMOS-Developer) [inline MaterialBanner/src/main/ets/components/MainPage/MainPage.ets]: handle null & undefined validation for buttonAction1 callback function

File: MaterialBanner/src/main/ets/components/MainPage/MainPage.ets [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+              Text(this.bModel.getBannerMessage())
+                .fontSize(this.bModel.getTextSize()).fontFamily(this.bModel.getTextStyle())
+                .padding({ top: '10vp', left: '16vp' }).width('100%')
+            }.height('40vp').width('300vp')
+          }
+          Flex({ direction: FlexDirection.RowReverse, alignItems: ItemAlign.Start, justifyContent: FlexAlign.SpaceBetween }) {
+            Button(this.bModel.getButtonText2(), { type: ButtonType.Normal, stateEffect: true }).borderRadius('5vp')
+              .backgroundColor(this.bModel.getButtonColor())
+              .onClick((event: ClickEvent) => {
+                this.buttonAction2()
```
Comment (HMOS-Developer) [inline MaterialBanner/src/main/ets/components/MainPage/MainPage.ets]: handle null & undefined validation for buttonAction2 callback function

File: MaterialBanner/src/main/ets/components/MainPage/MainPage.ets [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+                this.bModel.setShow(Visibility.None)
+              })
+            Button(this.bModel.getButtonText1(), { type: ButtonType.Normal, stateEffect: true }).borderRadius('5vp')
+              .backgroundColor(this.bModel.getButtonColor())
+              .onClick((event: ClickEvent) => {
+                this.buttonAction1()
```
Comment (HMOS-Developer) [inline MaterialBanner/src/main/ets/components/MainPage/MainPage.ets]: handle null & undefined validation for buttonAction1 callback function

File: MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+    return this.image
+  }
+
+  setButtonColor(value:string|ResourceColor):BannerModel{
+    if(value==null || value==undefined){
+      console.log("Button color not set. Default value used")
+    }
+    else {
+      this.buttonColor = value
+      return this
```
Comment (Himanshu-0123) [inline MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets]: return statement should be after the else block

File: MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+    }
+  }
+  setTextSize(tc:string):BannerModel{
+    if(tc==null || tc==undefined){
+      console.log("Text size not set. Default value used")
+    }
+    else {
+      this.textSize = tc
+      return this
```
Comment (Himanshu-0123) [inline MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets]: return statement should be after the else block

File: MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+    }
+  }
+  setTextStyle(ts:string):BannerModel{
+    if(ts==null || ts==undefined){
+      console.log("Text style not set. Default value used")
+    }
+    else {
+      this.textStyle = ts
+      return this
```
Comment (Himanshu-0123) [inline MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets]: return statement should be after the else block

File: MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+    }
+  }
+  setButtonText1(a1:string):BannerModel{
+    if(a1==null || a1==undefined){
+      console.log("Button Text 1 not set")
+    }
+    else {
+      this.buttonText1 = a1
+      return this
```
Comment (Himanshu-0123) [inline MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets]: return statement should be after the else block

File: MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+  setButtonText2(a2:string):BannerModel{
+    if(this.bannerType==BannerType.TwoButton && (a2==null || a2==undefined)){
+      console.log("Button Text 2 not set")
+    }
+    else if(this.bannerType==BannerType.ImageButton && (a2==null || a2==undefined)){
+      console.log("Button Text 2 not set")
+    }
+    else {
+      this.buttonText2 = a2
+      return this
```
Comment (Himanshu-0123) [inline MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets]: return statement should be after the else block

File: MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+    }
+  }
+  setImage(img:Resource):BannerModel{
+    if(this.bannerType==BannerType.ImageButton && (img==null || img==undefined)){
+      console.log("Image not selected")
+    }
+    else {
+      this.image = img
+      return this
```
Comment (Himanshu-0123) [inline MaterialBanner/src/main/ets/components/MainPage/BannerModel.ets]: return statement should be after the else block

**Comment by Himanshu-0123** [comment]
add copyright text in all js/ets/ts files in test folder also

File: MaterialBanner/hvigorfile.js [hunk 20]
```diff
@@ -0,0 +1,3 @@
+// Script for compiling build behavior. It is built in the build plug-in and cannot be modified currently.
```
Comment (Himanshu-0123) [inline MaterialBanner/hvigorfile.js:16]: add copyright text

File: entry/hvigorfile.js [hunk 21]
```diff
@@ -0,0 +1,2 @@
+// Script for compiling build behavior. It is built in the build plug-in and cannot be modified currently.
```
Comment (Himanshu-0123) [inline entry/hvigorfile.js:16]: add copyright text

File: hvigorfile.js [hunk 22]
```diff
@@ -0,0 +1,2 @@
+// Script for compiling build behavior. It is built in the build plug-in and cannot be modified currently.
```
Comment (Himanshu-0123) [inline hvigorfile.js:16]: add copyright text

File: MaterialBanner/src/main/ets/components/MainPage/MainPage.ets [hunk 23]
```diff
@@ -0,0 +1,105 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {BannerModel,BannerType} from './BannerModel'
+
+@Component
+export struct MaterialBanner{
+  @State bModel:BannerModel=new BannerModel(BannerType.OneButton);
```
Comment (archanaatr) [inline MaterialBanner/src/main/ets/components/MainPage/MainPage.ets]: format all the ets files, perform ctrl + alt + L in studio.

File: README.md [hunk 24]
```diff
@@ -0,0 +1,121 @@
...
+  this.bannerModel3.setImage(this.image)
+}
+```
+
+```
+//Passing custom object to MaterialBanner
+MaterialBanner({  
+  bModel: this.bannerModel3,  
+  buttonAction1: this.BannerFunc3,  
+  buttonAction2: this.BannerFunc4
```
Comment (archanaatr) [inline README.md:111]: Please add screenshots or a gif to display about the component behavior

File: MaterialBanner/src/main/ets/components/MainPage/MainPage.ets [hunk 25]
```diff
@@ -0,0 +1,114 @@
...
+          }
+          .height('50vp')
+          .width('90%')
+          .padding({ top: '12vp', bottom: '8vp', left: '48vp' })
+        }.width('95%').visibility(this.bModel.getShow())
+      }
+      else if (this.bModel.getBannerType() == BannerType.ImageButton) {
+        Column() {
+          Flex({ justifyContent: FlexAlign.SpaceBetween }) {
+            Image(this.bModel.getImage()).height('60vp').width('60vp').padding({ left: '10vp', top: '6vp' })
```
Comment (archanaatr) [inline MaterialBanner/src/main/ets/components/MainPage/MainPage.ets:77]: click callback required for ImageButton? Please recheck.

File: entry/src/main/ets/pages/index.ets [hunk 26]
```diff
@@ -0,0 +1,128 @@
...
+  @State image: Resource= $r('app.media.noc')
+  @State display1: boolean = false
+  @State display2: boolean = false
+  @State display3: boolean = false
+  private bannerModel1: BannerModel= new BannerModel(BannerType.OneButton)
+  private bannerModel2: BannerModel= new BannerModel(BannerType.TwoButton)
+  private bannerModel3: BannerModel= new BannerModel(BannerType.ImageButton)
+
+  aboutToAppear() {
+    this.bannerModel1.setBannerMessage("Your Device is Offline");
```
Comment (archanaatr) [inline entry/src/main/ets/pages/index.ets:30]: Place all the strings at string.json and add localization also for chinese language (can translate strings online and place)

**Pull request merged**